### PR TITLE
[WIP] Provide a signal for opening items from a DirListing

### LINKED ISSUE #6269: No event fired when item clicked in FileBrowser
I'm trying to see how I can get notified from the FileBrowser component when a file is double clicked. Unfortunately, there seems to be no way to do this. The only notification mechanism available is via DocumentRegistry, which also assumes that you're going to create a new widget with it. 

Would love to have a callback that notifies it whenever an item is double clicked.

### PR DESCRIPTION
**Pull request by yuvipanda**
Possible way to fix #6269

### CODE REVIEW

File: packages/filebrowser/src/listing.ts [hunk 1]
```diff
@@ -897,6 +913,17 @@ export class DirListing extends Widget {
     this._startDrag(data.index, event.clientX, event.clientY);
   }
 
+  private emitOpenEvent(item: Contents.IModel): void {
```
Comment (blink1073) [inline packages/filebrowser/src/listing.ts]: Private methods should have a leading underscore.  Also, this does more than emit a signal.  How about:

```typescript
/**
 * Open a file item and emit a signal.
 */
private _openItem(item: Contents.IModel): void {
```

**Comment by jasongrout** [comment]
Does this prevent the item from opening locally? I thought you wanted to take over the opening, not just have a signal.

**Comment by jasongrout** [comment]
(from the formatting changes, it seems that you have an older version of prettier installed. I would check in any changes, `git clean -dfx` your repo directory, and reinstall to get a fresh set of dependencies including an updated `prettier`)

**Comment by blink1073** [comment]
Made a few tweaks, this is good to go, thanks @yuvipanda!

**Comment by yuvipanda** [comment]
Thanks, @blink1073!

**Pull request merged**